File: output/GPT3.5-Turbo/xCodeEval_remove_4000/Java/Python/fcb7f4723da8f04efdfa91be61afe67a.py
import sys
from typing import List, Tuple, Dict

class Count:
    def __init__(self):
        self.count = 0
        self.cost = 0

    def increment(self, cost: int) -> None:
        if self.count >= k:
            return
        self.cost += cost
        self.count += 1

def solver() -> None:
    n, k = map(int, input().split())
    arr = list(map(int, input().split()))
    arr.sort()
    counts: Dict[int, Count] = {0: Count()}
    for number in arr:
        num = number
        cost = 0
        while num > 0:
            if num not in counts:
                counts[num] = Count()
            counts[num].increment(cost)
            num //= 2
            cost += 1
        counts[0].increment(cost)
    ans = sys.maxsize
    for count in counts.values():
        if count.count >= k:
            ans = min(ans, count.cost)
    print(ans)

if __name__ == '__main__':
    solver()

File: output/GPT4o-Mini/xCodeEval_remove_4000/Java/C/fcb7f4723da8f04efdfa91be61afe67a.c
#include <stdio.h>
#include <stdlib.h>
#include <limits.h>
#include <string.h>

typedef struct {
    int count;
    long cost;
} Count;

typedef struct {
    int *data;
    int size;
} IntArray;

typedef struct {
    IntArray *counts;
    int k;
} Solver;

void increment(Count *count, long cost) {
    if (count->count >= count->k) return;
    count->cost += cost;
    count->count++;
}

long min(long a, long b) {
    return (a < b) ? a : b;
}

void sort(IntArray *arr) {
    qsort(arr->data, arr->size, sizeof(int), (int (*)(const void *, const void *))compare);
}

int compare(const void *a, const void *b) {
    return (*(int *)a - *(int *)b);
}

void solver(Solver *solver) {
    int n;
    scanf("%d", &n);
    scanf("%d", &solver->k);
    
    IntArray arr;
    arr.size = n;
    arr.data = (int *)malloc(n * sizeof(int));
    
    for (int i = 0; i < n; i++) {
        scanf("%d", &arr.data[i]);
    }
    
    sort(&arr);
    
    Count *counts = (Count *)calloc(1000001, sizeof(Count)); // Assuming max value is 1000000
    counts[0].count = 0;
    long ans = LONG_MAX;
    
    for (int i = 0; i < arr.size; i++) {
        int num = arr.data[i];
        int cost = 0;
        while (num > 0) {
            if (num < 1000001 && counts[num].count == 0) {
                counts[num].count = 0;
                counts[num].cost = 0;
            }
            increment(&counts[num], cost);
            num /= 2;
            cost++;
        }
        increment(&counts[0], cost);
    }
    
    for (int i = 0; i < 1000001; i++) {
        if (counts[i].count >= solver->k) {
            ans = min(ans, counts[i].cost);
        }
    }
    
    printf("%ld\n", ans);
    
    free(arr.data);
    free(counts);
}

int main() {
    Solver solver;
    solver.k = 0;
    solver.counts = NULL;
    solver(&solver);
    return 0;
}
